modules/images/regenerate-existing-images/helper.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Functions used by the background process API.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Creates a background job.
+ *
+ * Job refers to the term and queue refers to the `background_job` taxonomy.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $name Name of job identifier.
+ * @param array  $data Data for the job.
+ * @return Perflab_Background_Job|WP_Error Job object if created successfully, else WP_Error.
+ */
+function perflab_create_background_job( $name, array $data = array() ) {
+	// Insert the new job in queue.
+	$term_name = 'job_' . time() . rand();
+	$term_data = wp_insert_term( $term_name, PERFLAB_BACKGROUND_JOB_TAXONOMY_SLUG );
+
+	if ( ! is_wp_error( $term_data ) ) {
+		update_term_meta( $term_data['term_id'], Perflab_Background_Job::META_KEY_JOB_DATA, $data );
+		update_term_meta( $term_data['term_id'], Perflab_Background_Job::META_KEY_JOB_NAME, $name );
+
+		// Create a fresh instance to return.
+		$job = new Perflab_Background_Job( $term_data['term_id'] );
+		// Set the queued status for freshly created jobs.
+		$job->set_status( Perflab_Background_Job::JOB_STATUS_QUEUED );
```

Comment:
I'm not strongly opposed to setting this here, but if this is set for every new job term, shouldn't we also make it a default value in case no status is set in the database for a job? I think it would make sense to incorporate that default into the `Perflab_Background_Job::get_status()` method (if the term meta value is empty).